great_expectations/datasource/new_datasource.py

```diff
@@ -59,14 +61,16 @@ def __init__(
                 runtime_environment={"concurrency": concurrency},
                 config_defaults={"module_name": "great_expectations.execution_engine"},
             )
-            self._datasource_config: dict = {
-                "execution_engine": execution_engine,
-                "id": id,
-                "name": name,
-            }
         except Exception as e:
             raise ge_exceptions.ExecutionEngineError(message=str(e))
 
+        self._datasource_config: dict = {
+            "execution_engine": execution_engine,
+            "id": id,
+            "name": name,
+        }
```

Comment:
The `try/except` was for the `ExecutionEngine` creation - no need to put this `self._datasource_config` creation in the same block.